source/server/listener_manager_impl.cc

```diff
@@ -281,6 +297,9 @@ void ListenerManagerImpl::drainListener(ListenerImplPtr&& listener) {
     std::lock_guard<std::mutex> guard(draining_listeners_lock_);
     draining_it = draining_listeners_.emplace(draining_listeners_.begin(), std::move(listener),
                                               workers_.size());
+    // Must be done under lock. Set is used so hot restart / multiple processes converge to a
+    // single value. Same below inside the lambda.
+    stats_.total_listeners_draining_.set(draining_listeners_.size());
```

Comment:
Instead of saying "converge" it might be clearer if the comment just says that using set() avoids multiple modifiers problem during the multiple processes phase of hot restart.